store/tikv/client.go

```diff
@@ -50,9 +50,10 @@ var MaxRecvMsgSize = math.MaxInt64
 // Timeout durations.
 var (
 	dialTimeout               = 5 * time.Second
-	readTimeoutShort          = 20 * time.Second  // For requests that read/write several key-values.
-	ReadTimeoutMedium         = 60 * time.Second  // For requests that may need scan region.
-	ReadTimeoutLong           = 150 * time.Second // For requests that may need scan region multiple times.
+	readTimeoutShort          = 20 * time.Second   // For requests that read/write several key-values.
+	ReadTimeoutMedium         = 60 * time.Second   // For requests that may need scan region.
+	ReadTimeoutLong           = 150 * time.Second  // For requests that may need scan region multiple times.
+	ReadTimeoutTiFlash        = 3600 * time.Second // For requests that may scan many regions for tiflash.
```

Comment:
Any chance we don't hardcode TiFlash here? I suggest use things like ultra long.